packages/headless/src/features/insight-interface/insight-interface-state.ts

```diff
@@ -0,0 +1,16 @@
+import {InsightInterface} from '../../api/service/insight/get-interface/get-interface-config-response';
+import {InsightAPIErrorStatusResponse} from '../../api/service/insight/insight-api-client';
+
+export const getInsightInterfaceInitialState = (): InsightInterfaceState => ({
+  loading: false,
+  config: undefined,
+});
+
+export interface InsightInterfaceState {
+  loading: boolean;
+  config?: {
```

Comment:
I think a better type would be to have an always defined/available `config`, but with a bunch of empty properties/object in `getInsightInterfaceInitialState().config`.

Its otherwise kind of annoying for a user of Headless to have to add defensive null ref check in a bunch of place in their code.

Reply:
In this case, knowing whether the interface config is loaded is very important because the container component has to wait for `config` to be filled before it can add the search components to the search interface. So, in my opinion, having a nullable `config` makes it obvious.